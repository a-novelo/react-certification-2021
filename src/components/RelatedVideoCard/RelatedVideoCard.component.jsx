import React from 'react';
import { useHistory } from 'react-router-dom';
import {
  RelatedVideoCardWrapper,
  RelatedVideoCardButton,
  RelatedVideoCardImage,
  RelatedVideoCardBody,
  RelatedVideoCardTitle,
} from './RelatedVideoCard.styled';

function RelatedVideoCard(props) {
  const history = useHistory();
  const { photoHeader, title, videoId } = props;

  return (
    <RelatedVideoCardWrapper>
      <RelatedVideoCardButton
        onClick={() => {
          history.push(`/video/${videoId}`);
        }}
      >
        <RelatedVideoCardImage src={photoHeader} />
        <RelatedVideoCardBody>
          <RelatedVideoCardTitle>{title}</RelatedVideoCardTitle>
        </RelatedVideoCardBody>
      </RelatedVideoCardButton>
    </RelatedVideoCardWrapper>
  );
}

export default RelatedVideoCard;
