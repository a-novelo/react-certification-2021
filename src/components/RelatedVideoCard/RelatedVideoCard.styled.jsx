import styled from 'styled-components';

const RelatedVideoCardWrapper = styled.div`
  border: 1px solid silver;
  height: 75px;
  overflow: hidden;
  width: 100%;
`;

const RelatedVideoCardButton = styled.button`
  align-items: center;
  background: none;
  border: none;
  color: inherit;
  cursor: pointer;
  display: flex;
  font: inherit;
  outline: inherit;
  padding: 0;
`;

const RelatedVideoCardImage = styled.img`
  height: 100%;
  width: 100px;
  object-fit: cover;
`;

const RelatedVideoCardBody = styled.div`
  text-align: start;
  padding: 16px;
`;

const RelatedVideoCardTitle = styled.div`
  font-size: 16px;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 500;
  line-height: 1.6;
  letter-spacing: 0.0075em;
`;

export {
  RelatedVideoCardWrapper,
  RelatedVideoCardButton,
  RelatedVideoCardImage,
  RelatedVideoCardBody,
  RelatedVideoCardTitle,
};
