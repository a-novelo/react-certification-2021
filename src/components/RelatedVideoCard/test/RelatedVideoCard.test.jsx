import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import RelatedVideoCard from '../RelatedVideoCard.component';

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('#RelatedVideoCard', () => {
  it('displays correctly', () => {
    const tree = renderer.create(<RelatedVideoCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has button to navigate to video', () => {
    render(<RelatedVideoCard />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(mockHistoryPush).toHaveBeenCalledWith('/video/undefined');
  });
});
