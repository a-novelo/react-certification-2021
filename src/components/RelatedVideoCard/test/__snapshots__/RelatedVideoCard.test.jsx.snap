// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#RelatedVideoCard displays correctly 1`] = `
<div
  className="sc-bdfBwQ iMqYPQ"
>
  <button
    className="sc-gsTCUz fGQDbM"
    onClick={[Function]}
  >
    <img
      className="sc-dlfnbm kCQBmk"
    />
    <div
      className="sc-hKgILt jnTFzE"
    >
      <div
        className="sc-eCssSg fPOxlr"
      />
    </div>
  </button>
</div>
`;
